scripts/analyze_csv_data.py

```diff
@@ -481,19 +483,22 @@ def performance_ratio_fn(df, par):
         df.groupby(["instance_name"])
         .apply(lambda _: performance_ratio_fn(_, par))
         .reset_index(name="ratio")
+        .dropna()
     )
     plt.figure(figsize=(10, 6))
     plt.title(
         sanitize_title(
             f"{prefix} {xlabel} {label_lookup(l0)}:{label_lookup(l1)}"
         )
     )
-    plot_loghist(ratios["ratio"], min(len(ratios) // 3, 25))
-    path = os.path.join(
-        FIGS_DIR,
-        f"{prefix}_{label_lookup(l0)}_{label_lookup(l1)}_{xaxis}_performance_ratio.pdf",
-    )
-    plt.savefig(path)
+    nbins = min(len(ratios) // 3, 25)
+    if nbins > 0:
+        plot_loghist(ratios["ratio"], nbins)
+        path = os.path.join(
+            FIGS_DIR,
+            f"{prefix}_{label_lookup(l0)}_{label_lookup(l1)}_{xaxis}_performance_ratio.pdf",
```

Comment:
pre-existing nit: since this is the only line that is > 80 characters, and this PR is touching it anyway, break the string into two concatenated pieces so it can fit in 80-character lines.